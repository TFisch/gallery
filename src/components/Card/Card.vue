<template>
  <div class="card">
    <img v-bind:src="this.painting.image[0].baseimageurl" class="image" />
  </div>
</template>

<script>
export default {
  name: 'card',
  props: ['painting']
};
</script>

<style scoped>
.card {
  align-items: center;
  background-color: magenta;
  border: solid black 1px;
  display: flex;
  height: 20em;
  justify-content: center;
  margin-left: auto;
  margin-right: auto;
  margin-top: 1em;
  width: 16em;
  overflow: hidden;
}

.image {
  align-items: center;
  width: auto;
  height: auto;
  max-width: 100%;
  max-height: 100%;
}
</style>