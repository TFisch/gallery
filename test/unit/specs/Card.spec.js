import { shallowMount } from '@vue/test-utils';
import Card from './../../../src/components/Card/Card';

describe('Card.vue', () => {
  let wrapper;
  let mockPainting;

  it('should match th snapshot', () => {
    mockPainting = { image: 'google.com' };
    wrapper = shallowMount(Card, {
      propsData: {
        painting: mockPainting
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
