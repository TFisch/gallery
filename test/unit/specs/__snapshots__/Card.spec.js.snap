// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue should match th snapshot 1`] = `"<div class=\\"card\\"><img src=\\"google.com\\" class=\\"image\\"></div>"`;
